### ISSUE TITLE
Finally I reproduce the result on the same wild skating video

### ISSUE THREAD

**Issue by bucktoothsir** [facebookresearch/VideoPose3D#23]
![camera1](https://user-images.githubusercontent.com/1781061/51375886-f6443480-1b41-11e9-86f2-9465d457102f.gif)

**Comment by bucktoothsir** [comment]
![image](https://user-images.githubusercontent.com/1781061/51376588-a7979a00-1b43-11e9-9cf8-2cd7962d27bc.png)
![image](https://user-images.githubusercontent.com/1781061/51376598-ab2b2100-1b43-11e9-807d-df5c9b4fa5ee.png)

**Comment by DongJT1996** [comment]
Hello,
It's cool to reproduce the result. May I ask you a question that have you tried to change the viewpoint to see the result?

**Comment by bucktoothsir** [comment]
@DongJT1996 yes I have

**Comment by tobiascz** [comment]
@bucktoothsir - We are discussing this topic in 3 different issues now 💃 

Really cool! What do you think is the reason for this improvement? 
1. Smaller Image so that the dancer is always more in  the center and detectron estimates the 2D positions more accurate? 
2. Something  internal in the VideoPose3D code? 
3. Your computer was in a good mood?

**Comment by bucktoothsir** [comment]
@tobiascz 
I think the reason is that the h36m dataset is square. 
So the model trained on h36m dataset could only works on square pictures.

**Comment by tobiascz** [comment]
> @tobiascz
> I think the reason is that the h36m dataset is square.
> So the model trained on h36m dataset could only works on square pictures.

But you don't train directly on the images. You train the network on 2D poses and that's just an 17,2 array for each frame. The network lifts the 2D poses to 3D and you can compare the results to the ground truth provided in Human 3.6m. The only one who cares about your image resolution or size or whatever is your 2D Detector and that is in our case detectron.

**Comment by bucktoothsir** [comment]
> > @tobiascz
> > I think the reason is that the h36m dataset is square.
> > So the model trained on h36m dataset could only works on square pictures.
> 
> But you don't train directly on the images. You train the network on 2D poses and that's just an 17,2 array for each frame. The network lifts the 2D poses to 3D and you can compare the results to the ground truth provided in Human 3.6m. The only one who cares about your image resolution or size or whatever is your 2D Detector and that is in our case detectron.

2D pose is relative to the size of image. check code 84 col in run.py
`kps[..., :2] = normalize_screen_coordinates(kps[..., :2], w=cam['res_w'], h=cam['res_h'])
`
but your thought might be right. 

it is easy to verify our guess. 
1.  
  1.1  extract 2d pose by original videos
  1.2 cut the videos to square and change 2d pose by translation.
  1.3. reconstruct 3d pose

2.
  2.1 cut the videos to square
  2.2 extract 2d pose on square videos
  2.3 reconstruct 3d pose

compare the two results you would find something.

**Comment by tobiascz** [comment]
@bucktoothsir  point 1. to 1.3 I already did [here](https://github.com/facebookresearch/VideoPose3D/issues/2#issuecomment-458022079) right.

So I tried what you suggested. I cutted the original video to square. Extracted 2D pose with detectron and reconstructed 3D pose using VideoPose3D.
## 2nd Method
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/8737489/51989761-047e4300-24a8-11e9-94f8-5720591685af.gif)

So yeah it still doesn't look as nice as yours.

**Comment by tobiascz** [comment]
The 2D keypoints metadata was my problem:
`keypoints_symmetry = [ [1, 3, 5, 7, 9 , 11, 13, 15],[2, 4, 6, 8, 10, 12, 14, 16]]`
with this symmetry I get really good results. Had to fix my 3D visualization as [mentioned](https://github.com/facebookresearch/VideoPose3D/issues/23#issuecomment-458975934) but now I am very happy with the reults! 

![scater_gif](https://user-images.githubusercontent.com/8737489/52120837-f0f4e880-261d-11e9-8827-77a869f37d6a.gif)

**Comment by dariopavllo** [comment]
Good catch! Indeed, the keypoint symmetry is used for test-time augmentation. If the latter is enabled (it is by default), the match between left/right keypoints should be specified properly to avoid messing things up.

Regarding previous comments: the image does not need to have a square aspect ratio. I made the videos using the original 1920x1080 resolution. The important thing is that the images are normalized so that the largest edge is between -1 and 1 and the center of the image is at (0, 0), but the normalization function `normalize_screen_coordinates` takes care of that already. You should just make sure to modify `cam['res_w']` and `cam['res_h']` in the dataset definition to reflect the actual resolution of the video.
